src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,287 @@
+      path.resolve(__dirname, '..', 'loaders'),
+      'node_modules',
+    ];
+
+    // generate the 404 page.
+    new HtmlWebpackPlugin({
+      filename: '404.html',
+      template: path.join(__dirname, '..', 'templates', '404.html'),
+      inject: 'head',
+      chunks: ['redirect'],
+    }).apply(compiler);
+
+    // WEBPACK DEV SERVER
+    if (compiler.options.devServer) {
+      // Setting this to enable browser routing
+      compiler.options.devServer.historyApiFallback = true;
+    }
+
+    new DefinePlugin({
+      // Base name is used to namespace terra-dev-site this is used in redirect.js which is only used in the 404 page.
+      TERRA_DEV_SITE_BASENAME: JSON.stringify(basename),
+    }).apply(compiler);
+  }
+
+  apply(compiler) {
+    // Use default public path else the env else /
+    let defaultPublicPath;
+    if (compiler.options.output && compiler.options.output.publicPath) {
+      defaultPublicPath = compiler.options.output.publicPath;
+    }
+    const publicPath = defaultPublicPath || process.env.TERRA_DEV_SITE_PUBLIC_PATH || '/';
+
+    // OUTPUT
+    compiler.options.output.publicPath = publicPath;
+
+    // Strip the trailing / from the public path.
+    let basename = publicPath.slice(0, -1);
+
+    const { sourceFolder, distributionFolder } = this.siteConfig;
+
+    // Since there can be multiple dev site plugins this config we only want to do once for all of them.
+    SitePlugin.applyOneTimeSetup({
+      compiler,
+      sourceFolder,
+      distributionFolder,
+      basename,
+    });
+
+    // ENTRY
+    compiler.options.entry = {
+      ...compiler.options.entry,
+      [this.entryKey]: `@cerner/terra-dev-site/lib/webpack/templates/entry.template${this.resourceQuery}`,
+    };
+
+    // Get the list of apps excluding this current app.
+    const filteredSites = Object.values(siteRegistry).filter(site => site.path !== this.siteConfig.pathPrefix);
+
+    // Map to what we want to send to site config
+    const otherSites = filteredSites.map((site) => ({
+      path: site.path,
+      title: site.title,
+      url: `${basename}${site.url}`,
+    }));
+
+    // if there is a path prefix we want to update the react router basename to include the prefix.
+    if (this.siteConfig.pathPrefix) {
+      basename = [basename, this.siteConfig.pathPrefix].join('/');
+    }
+
+    // MODULE
+    // we know there is a oneOf here because we just added it.
+    compiler.options.module.rules[0].oneOf.unshift({
+      // This loader generates the entrypoint and sets up the config template path and resource query.
+      resourceQuery: this.resourceQuery,
+      use: [
+        babelLoader,
+        {
+          loader: 'devSiteEntry',
+          options: {
+            entryPath: this.entry,
+            siteConfig: this.siteConfig,
+            sites: otherSites,
+            basename,
+            resolveExtensions: compiler.options.resolve.extensions,
+            isLernaMonoRepo,
```

Comment:
I think you could remove the monorepo if-else path/pattern logic in generateNavigationConfig addWatchContentDirectories and getSearchPatterns if you pass down the rootDirectories or the resolves root path.

The `isLernaMonoRepo` is just passed down till it's used in generateNavigationConfig from what I can tell

Reply:
It's true, but I'm not sure it makes much of a difference performance wise. either way I still need to glob the directories and would still have to build out the entrypoint regex seprately.

Reply:
i also don't want to glob if it's not a mono repo directory